src/components/Conversation/ViewPoints/ContentCard.tsx

```diff
@@ -0,0 +1,126 @@
+"use client";
+import { useState, useRef, useEffect } from "react";
+import { ViewPoint } from "@/types/conversations.types";
+import { Spoiler } from "@mantine/core";
+import {
+    HandThumbUpIcon,
+    HandThumbDownIcon,
+    ArrowUpCircleIcon,
+} from "@heroicons/react/24/solid";
+import { ArrowDownIcon, ArrowUpIcon } from "@heroicons/react/24/outline";
+import Avatar from "../Avatar/Avatar";
+
+type ContentCardProps = {
+    viewpoint: ViewPoint;
+};
+
+enum ReactionStatus {
+    like = "like",
+    dislike = "dislike",
+    reasonable = "reasonable",
+    null = "null",
+}
+
+export default function ContentCard({ viewpoint }: ContentCardProps) {
+    const [reactionStatus, setReactionStatus] = useState<ReactionStatus>(
+        ReactionStatus.null,
+    );
+    const [showContentHeight, setShowContentHeight] = useState<number>(0);
+    const firstParagraphHeight = useRef<HTMLParagraphElement | null>(null);
+
+    useEffect(() => {
+        if (firstParagraphHeight.current) {
+            setShowContentHeight(firstParagraphHeight.current.clientHeight);
+        }
+    }, []);
+    const handleReaction = (reaction: ReactionStatus) => {
+        setReactionStatus((prev) =>
+            prev === reaction ? ReactionStatus.null : reaction,
+        );
+    };
+
+    return (
+        <div>
+            <div className="mb-1 flex">
+                <div className="inline">
+                    <Avatar user={viewpoint.user} />
+                </div>
+                <h1 className="ml-3 inline-block text-xs font-normal text-neutral-600">
+                    {viewpoint.created.toLocaleDateString()}
+                </h1>
+            </div>
+            <h1 className="text-lg font-semibold text-neutral-700">
+                {viewpoint.title}
+            </h1>
+            <Spoiler
+                maxHeight={showContentHeight}
+                showLabel={
+                    <span>
+                        查看完整觀點
+                        <ArrowDownIcon className="inline size-5 fill-none stroke-emerald-600 stroke-[1.5]" />
+                    </span>
+                }
+                hideLabel={
+                    <span>
+                        顯示較少
+                        <ArrowUpIcon className="inline size-5 fill-none stroke-emerald-600 stroke-[1.5]" />
+                    </span>
+                }
+                classNames={{ control: "text-emerald-600 underline" }}
+            >
+                <p
+                    key={0}
+                    ref={firstParagraphHeight}
+                    className="text-base font-normal text-black"
+                >
+                    {viewpoint.content.split("\n")[0]}
+                </p>
+                {viewpoint.content.split("\n").map(
```

Comment:
Consider doing filtering on the array front (e.g., with `content.split("\n").slice(1)`)